tensorflow/lite/kernels/activations.cc

```diff
@@ -363,23 +365,30 @@ TfLiteStatus LeakyReluPrepare(TfLiteContext* context, TfLiteNode* node) {
 
   LeakyReluOpData* data = reinterpret_cast<LeakyReluOpData*>(node->user_data);
 
-  if (output->type == kTfLiteUInt8) {
-    const auto* params =
-        reinterpret_cast<TfLiteLeakyReluParams*>(node->builtin_data);
-    // Quantize the alpha with same zero-point and scale as of input
-    data->q_alpha = static_cast<uint8_t>(std::max<float>(
-        std::numeric_limits<uint8_t>::min(),
-        std::min<float>(std::numeric_limits<uint8_t>::max(),
-                        std::round(input->params.zero_point +
-                                   (params->alpha / input->params.scale)))));
-
-    double real_multiplier =
-        input->params.scale * input->params.scale / output->params.scale;
-    QuantizeMultiplierSmallerThanOneExp(
-        real_multiplier, &data->output_multiplier, &data->output_shift);
-  }
-  return context->ResizeTensor(context, output,
-                               TfLiteIntArrayCopy(input->dims));
+  if (output->type == kTfLiteUInt8 || output->type == kTfLiteInt8) {
+    const auto* params = reinterpret_cast<TfLiteLeakyReluParams*>(node->builtin_data);
+    // Quantize the alpha with predetermined ZOOM_FACTOR and ZERO_POINT.
+    // Since in most cases 0 < alpha < 1, by setting ZOOM_FACTOR to be 200,
+    // quantized alpha can fit in INT8 range, and preserve most precision.
+    TF_LITE_ENSURE(context, params->alpha >= 0);
+    TF_LITE_ENSURE(context, params->alpha < 1);
+    static const uint8_t ZOOM_FACTOR = 200;
+    static const uint8_t ZERO_POINT = 0;
+    auto q_alpha = std::round(ZERO_POINT + params->alpha * ZOOM_FACTOR);
+    // Make sure quantized alpha is within INT8 range.
+    TF_LITE_ENSURE(context, q_alpha >= std::numeric_limits<uint8_t>::min());
+    TF_LITE_ENSURE(context, q_alpha <= std::numeric_limits<uint8_t>::max());
+    // q_alpha will be stored as uint8_t. It won't affect the input
+    data->q_alpha = q_alpha;
+    std::cout << "Quantized Q alpha value is: " << int(data->q_alpha) << '\n';
```

Comment:
can you check the coding style? also we don't need to log the value

Reply:
Yeah... It was used for debugging. I will delete that line. Thank you for catching that.